test/arbitrage/BancorArbitrage.ts

```diff
@@ -0,0 +1,358 @@
+        });
+
+        it('should be able to set and update the arbitrage rewards settings', async () => {
+            await bancorArbitrage.setRewards(ArbitrageRewardsDefaults);
+
+            const res = await bancorArbitrage.rewards();
+            expect(res.percentagePPM).to.equal(30_000);
+
+            const resChanged = await bancorArbitrage.setRewards(ArbitrageRewardsChanged);
+            await expect(resChanged).to.emit(bancorArbitrage, 'RewardsUpdated');
+
+            const resUpdated = await bancorArbitrage.rewards();
+            expect(resUpdated.percentagePPM).to.equal(40_000);
+        });
+    });
+
+    describe('arbitrage', () => {
+        beforeEach(async () => {
+            await transfer(deployer, bnt, masterVault.address, AMOUNT.mul(10_000));
+            await bancorArbitrage.setRewards(ArbitrageRewardsDefaults);
+        });
+
+        // get all exchange ids (omit their names)
+        const exchangeIds = Object.values(ExchangeId).filter((key) => !isNaN(Number(key)));
+        const uniV3Fees = [100, 500, 3000];
+        const tokenSymbols = [TokenSymbol.TKN1, TokenSymbol.TKN2, TokenSymbol.ETH];
+
+        // remove BancorV3 exchange until the reentrancy guard issue is resolved
+        exchangeIds.splice(exchangeIds.indexOf(ExchangeId.BancorV3), 1);
+
+        for (const exchangeId of exchangeIds) {
+            for (const arbToken1Symbol of tokenSymbols) {
+                for (const arbToken2Symbol of tokenSymbols) {
+                    if (arbToken1Symbol === arbToken2Symbol) {
+                        continue;
+                    }
+
+                    const exchangeName = ExchangeId[Number(exchangeId)];
+
+                    it(`arbitrage between ${arbToken1Symbol} and ${arbToken2Symbol} using ${exchangeName}`, async () => {
+                        const { token: arbToken1 } = await prepareBancorV3PoolAndToken(arbToken1Symbol);
+                        const { token: arbToken2 } = await prepareBancorV3PoolAndToken(arbToken2Symbol);
+
+                        let customInt;
+                        if (exchangeId === ExchangeId.UniswapV3) {
+                            customInt = uniV3Fees[tokenSymbols.indexOf(arbToken1Symbol)];
+
+                            await transfer(deployer, weth, exchanges.address, AMOUNT.mul(10));
+                        } else {
+                            customInt = 0;
+                        }
+
+                        await transfer(deployer, bnt, exchanges.address, AMOUNT.mul(10));
+                        await transfer(deployer, arbToken1, exchanges.address, AMOUNT.mul(10));
+                        await transfer(deployer, arbToken2, exchanges.address, AMOUNT.mul(10));
+
+                        const routes = [
+                            {
+                                exchangeId: ExchangeId.BancorV2,
+                                targetToken: arbToken1.address,
+                                minTargetAmount: 1,
```

Comment:
Minor: can use `getAmountsOut` here (and below) if you wanna make the test more restrictive.